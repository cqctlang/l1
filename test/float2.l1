@define mkpi(n)
{
	@local k, s, a;
	s = 0;
	a = 1;
	for(k = 0; k < n; k++){
		s += a/(float)(2*k+1);
		a *= -1;
	}
	return 4*s;
}
foreach(@lambda(n){ printf("%f\n", mkpi(n)); },
	[ 1, 10, 100, 1000, 10000 ]);
