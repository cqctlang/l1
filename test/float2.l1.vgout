4.000000
3.041840
3.131593
3.140593
3.141499
