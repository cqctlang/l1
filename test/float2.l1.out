entry: var params, 2 locals, 0 temps
mkpi: 1 params, 7 locals, 0 temps
L52: 1 params, 0 locals, 0 temps
4.000000
3.041840
3.131593
3.140593
3.141499
